Source/WebCore/rendering/RenderReplaced.cpp

```diff
@@ -634,8 +640,12 @@ LayoutUnit RenderReplaced::computeReplacedLogicalWidth(ShouldComputePreferred sh
             // on the replaced element's width, then the used value of 'width' is
             // calculated from the constraint equation used for block-level,
             // non-replaced elements in normal flow.
-            if (computedHeightIsAuto && !hasIntrinsicWidth && !hasIntrinsicHeight)
-                return computeConstrainedLogicalWidth(shouldComputePreferred);
+            if (computedHeightIsAuto && !hasIntrinsicWidth && !hasIntrinsicHeight) {
+                if (shouldComputePreferred == ComputePreferred)
+                    return computeReplacedLogicalWidthRespectingMinMaxWidth(0_lu, ComputePreferred);
+                auto constrainedLogicalWidth = computeConstrainedLogicalWidth();
+                return computeReplacedLogicalWidthRespectingMinMaxWidth(constrainedLogicalWidth, ComputeActual);
```

Comment:
I understand that you want to make `computeConstrainedLogicalWidth` do what it is saying it does, but don't you want to abstract this repeated logic in another lambda/function with a significant name?

Reply:
The constrained logical width computation should be completely separate from any preferred logical widths logic so factoring that out into the same function would be just as wrong. 

Aside from that we could refactor the constrained logical width computation and then the call to `computeReplacedLogicalWidthRespectingMinMaxWidth` into something but we already have similar functions like that on render tree and they are honestly confusing because it's really hard to tell when you should call which one :/